site/components/pages/Pledge/lib/firebase.ts

```diff
@@ -34,7 +53,11 @@ export const getPledgeByAddress = async (address: string): Promise<Pledge> => {
   const pledgeRef = pledgeSnapshot.docs[0];
 
   if (!pledgeRef) {
-    throw new Error("Bad request");
+    const e = new Error(
+      `We were unable to find a pledge for address: ${address}`
+    );
+    e.name = "PledgeNotFound";
```

Comment:
Something I'm realizing right now, we should document these, preferably with typescript. This can be done with:
1. a union type of all the names
```ts
type NamedError: "PledgeNotFound" | "WalletAlreadyPinned" | ...
```
2. use that union type in the expected return value for the API endpoint here https://github.com/KlimaDAO/klimadao/pull/729/files#diff-c9e7c84c661df8d858eda48e157a65da2c0216d9aac18a242face4cb77162965R6

more type safe, always better imo, but this may have waterfall effect of type errors, so if it causes issues we can do it in a follow-up tech debt ticket. In that case just make sure to document all of the known Named errors in a github ticket and link it here in this comment 👇